plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/cache/CachingHiveMetastore.java

```diff
@@ -494,23 +495,27 @@ public Map<String, PartitionStatistics> getPartitionStatistics(Table table, List
                 partitionsByName.keySet(),
                 missingPartitions -> loadPartitionsColumnStatistics(table, partitionsByName, missingPartitions),
                 currentStats -> currentStats.getColumnStatistics().keySet().containsAll(dataColumns),
-                CachingHiveMetastore::mergePartitionColumnStatistics);
+                (currentStats, newStats) -> mergePartitionColumnStatistics(currentStats, newStats, dataColumns));
         return statistics.entrySet().stream()
                 .collect(toImmutableMap(entry -> entry.getKey().getPartitionName().orElseThrow(), Entry::getValue));
     }
 
-    private static PartitionStatistics mergePartitionColumnStatistics(PartitionStatistics currentStats, PartitionStatistics newStats)
+    private static PartitionStatistics mergePartitionColumnStatistics(PartitionStatistics currentStats, PartitionStatistics newStats, Set<String> dataColumns)
     {
         requireNonNull(newStats, "newStats is null");
-        if (currentStats == null) {
-            return newStats;
-        }
+        ImmutableMap.Builder<String, HiveColumnStatistics> columnStatisticsBuilder = ImmutableMap.<String, HiveColumnStatistics>builder()
+                // Populate empty statistics for all requested columns to cache absence of column statistics for future requests.
+                // cacheMissing flag is ignored here as fetching column statistics is expensive and presence of statistics is not necessary for query to succeed.
```

Comment:
I'm still undecided about whether `cacheMissing` should be honoured here (the code change is 2 lines).
If the ETL process implements stats-on-write, then we're likely to see the new table/partition with stats from the beginning unless we're unlucky with the timing (read between table/partition creation and stats update in metastore).
I'm mainly concerned about a scenario where someone enables `cacheMissing` in a multiple cluster/engine environment to avoid object-not-found errors due to metadata cache being stale. But then they also incur overhead of repeatedly querying column stats unless they run ANALYZE (if we implement `cacheMissing` for column stats). 
@findepi wdyt ?

Reply:
> I'm mainly concerned about a scenario where someone enables cacheMissing in a multiple cluster/engine environment to avoid object-not-found errors due to metadata cache being stale.

Maybe we should have a separate flag for statistics? But in reality I would just honor the flag and educate users.